cirq/google/vis/heatmap.py

```diff
@@ -0,0 +1,250 @@
+    return rgb.dot([.2126, .7152, .0722])
+
+
+class Heatmap:
+    """Distribution of a value in 2D qubit lattice as a color map."""
+
+    def __init__(self, value_map: ValueMap) -> None:
+        self.set_value_map(value_map)
+        self.unset_annotation()
+        self.unset_url_map()
+        self.set_colorbar()
+        self.set_colormap()
+
+    def set_value_map(self, value_map: ValueMap) -> 'Heatmap':
+        """Sets the values for each qubit.
+
+        Args:
+            value_map: the values for determining color for each cell.
+        """
+        # Fail fast if float() fails.
+        # Keep the original value object for annotation.
+        self.value_map = {
+            key: (float(value), value) for key, value in value_map.items()
+        }
+        return self
+
+    def set_annotation_map(self, annot_map: Mapping[QubitCoordinate, str],
+                           **kws: str) -> 'Heatmap':
+        """Sets the annotation text for each qubit.
+
+        Note that set_annotation_map() and set_annotation_format()
+        both sets the annotation map to be used. Whichever is called later wins.
+
+        Args:
+            annot_map: the texts to be drawn on each qubit cell.
+            kws: keyword arguments to matplotlib.text.Text().
+        """
+        self.annot_map = annot_map
+        self.annot_kws = kws
+        return self
+
+    def set_annotation_format(self, annot_format: str, **kws: str) -> 'Heatmap':
+        """Sets a format string to format values for each qubit.
+
+        Args:
+            annot_format: the format string for formating values.
+            kws: keyword arguments to matplotlib.text.Text().
+        """
+        self.annot_map = {
+            key: format(value[1], annot_format)
+            for key, value in self.value_map.items()
+        }
+        self.annot_kws = kws
+        return self
+
+    def unset_annotation(self) -> 'Heatmap':
+        """Disables annotation. No texts are shown in cells."""
+        self.annot_map = {}
+        return self
+
+    def set_url_map(self, url_map: Mapping[QubitCoordinate, str]) -> 'Heatmap':
+        """Sets the URLs for each cell."""
+        self.url_map = url_map
+        return self
+
+    def unset_url_map(self) -> 'Heatmap':
+        """Disables URL. No URLs are associated with cells."""
+        self.url_map = {}
+        return self
+
+    def set_colorbar(self,
+                     position: str = 'right',
+                     size: str = '5%',
+                     pad: str = '2%',
+                     **kws: Any) -> 'Heatmap':
+        """Sets location and style of colorbar.
+
+        Args:
+            position: colorbar position, one of 'left'|'right'|'top'|'bottom'.
+            size: a string ending in '%' to specify the width of the colorbar.
+                Nominally, '100%' means the same width as the heatmap.
+            pad: a string ending in '%' to specify the space between the
+                colorbar and the heatmap.
```

Comment:
document what the kwargs are passed to.  Consider maybe renaming naming kwargs  to colobar_options

Reply:
Comments added.

Renamed all kwargs to foo_options (colorbar_options, text_options, pcolor_options).